### PR TITLE
chore(elixir): move rust vault to e folder

### PR DESCRIPTION
**Pull request by SanjoDeundiak**

### CODE REVIEW

**Comment by mrinalwadhwa** [comment]
@SanjoDeundiak something got lost in communication here. The mix package I pointed to in [my comment](https://github.com/ockam-network/ockam/pull/387#issuecomment-687247826) the other day is at 
https://github.com/ockam-network/ockam/tree/develop/implementations/e/applications/ockam_vault_software

The build is already setup in that package and there are couple of the NIFs Mike initially implemented already added. 

Sorry about the confusion.

**Comment by SanjoDeundiak** [comment]
@mrinalwadhwa fixed. I'll squash the commits after you verify everything is now in the right place and I rebase my other branches

File: implementations/e/applications/ockam_vault_software/lib/vault_software.ex [hunk 1]
```diff
@@ -1,48 +1,14 @@
...
-
   def load_natively_implemented_functions do
-    [:code.priv_dir(:ockam_vault_software), "native", "libockam_elixir_vault_software"]
+    [:code.priv_dir(:ockam), "native", "libockam_elixir_vault_software"]
     |> Path.join()
     |> to_charlist()
     |> :erlang.load_nif(0)
   end
 
-  def default_init do
```
Comment (mrinalwadhwa) [inline implementations/e/applications/ockam_vault_software/lib/vault_software.ex:45]: the highlighted code should stay the same as it was before this PR and only the new functions added.

**Review by mrinalwadhwa — APPROVED** [review]

**Review by mrinalwadhwa — CHANGES_REQUESTED** [review]
the rest of code looks good we can merge after the above highlighted section is added back

File: implementations/e/applications/ockam_vault_software/lib/vault_software.ex (hunk 1 above)
Comment (SanjoDeundiak) [reply implementations/e/applications/ockam_vault_software/lib/vault_software.ex]: Fixed

**Review by mrinalwadhwa — APPROVED** [review]

**Comment by mrinalwadhwa** [comment]
I made some minor changes to the commit message and merged b06b0a23da64068a7fbd921f5b87cad749a374ee
Could you rebase the other PRs to that.

**Pull request closed without merging**